api/queries_sponsors_test.go

```diff
@@ -0,0 +1,118 @@
+package api
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cli/cli/v2/pkg/httpmock"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_querySponsors(t *testing.T) {
+	var tests = []struct {
+		name           string
+		httpStubs      func(*httpmock.Registry)
+		expectedResult SponsorQuery
+		expectError    bool
+	}{
+		{
+			name: "success",
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query SponsorsList`),
+					httpmock.StatusStringResponse(200, `{"data": {"user": {"sponsors": {"totalCount": 2, "nodes": [{"login": "mona"}, {"login": "lisa"}]}}}}`))
+			},
+			expectedResult: SponsorQuery{
+				User: User{
+					Sponsors: Sponsors{
+						TotalCount: 2,
+						Nodes: []Node{
+							{Login: "mona"},
+							{Login: "lisa"},
+						},
+					},
+				},
+			},
+			expectError: false,
+		},
+		{
+			name: "error",
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query SponsorsList`),
+					httpmock.StatusStringResponse(500, `Internal Server Error`))
+			},
+			expectedResult: SponsorQuery{},
+			expectError:    true,
+		},
+	}
+
+	queryFnData := []struct {
+		name    string
+		queryFn func(*Client, string, string) (SponsorQuery, error)
+	}{
+		{
+			name:    "querySponsorsViaReflection",
+			queryFn: querySponsorsViaReflection,
```

Comment:
This is a clever use of injecting functions into the tests and it's nice you found it.

Reply:
Ideally, we wouldn't be testing private functions here. I suspect you're doing it for pragmatism as you try out the different approaches but I just want to call it out as something that I would be looking to avoid in real codebase. There's a few times where testing of private functions has impacted my ability to refactor code.

Reply:
That said, I do see that you have tested `formatTTYOutput` and `mapQueryToSponsorsList` as well, so maybe this is your preferred approach right now?